gamemodes/cinema/gamemode/modules/scoreboard/cl_playerlist.lua

```diff
@@ -203,6 +216,16 @@ function PLAYER:UpdatePlayer()
 			self.Mute:SetImage( "icon32/unmuted.png" )
 		end
     end
+	
+	self.ChatMute.DoClick = function() 
+		self.Player.IsChatMuted = !self.Player.IsChatMuted or true
```

Comment:
still doesn't work. if IsChatMuted is true, then !IsChatMuted is false, and the "or" will choose true. Just do IsChatMuted=not IsChatMuted

Reply:
but IsChatMuted would be null initially

Reply:
the not operator has higher precedence than the or operator, so its (not IsChatMuted) or true which will always equal true